src/main/kotlin/fi/aalto/cs/apluscourses/services/APlusModuleDeleteProvider.kt

```diff
@@ -0,0 +1,54 @@
+package fi.aalto.cs.apluscourses.services
+
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.module.ModuleManager
+import com.intellij.openapi.module.UnloadedModuleDescription
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.rootManager
+import com.intellij.openapi.roots.ui.configuration.actions.ModuleDeleteProvider
+import com.intellij.util.application
+import fi.aalto.cs.apluscourses.services.course.CourseManager
+import org.jetbrains.annotations.ApiStatus
+
+@Service(Service.Level.APP)
+internal class APlusModuleDeleteProvider : ModuleDeleteProvider() {
+    /**
+     * When removing a module from an A+ Courses project,
+     * it should not be deleted from other modules' dependencies
+     * because it would not get re-added when the module is re-downloaded.
+     */
+    @ApiStatus.Experimental
+    override fun doDetachModules(
```

Comment:
In the past all module dependencies were added to a json file and checked when the project opens. This way seems more robust and simple, but it could be tested a bit more